### DISCUSSION TITLE
Custom confirm delete message

### DISCUSSION THREAD

**Discussion by fractalbit** [Laravel-Backpack/community-forum#271] [Q&A (Help)] [answered]
Hello, quick question:

Is it possible to customize the confirmation message when deleting a crud resource depending on the resource?

I know i can change the translation for all resources, but can i display a different message depending on the resource being deleted? Can i set this, for example, in the crud controller for the resource? I can't seem to find the answer in the docs or by googling. Any help would be greatly appreciated.

**Comment by adamhaley** [comment]
I'm currently trying to do this too..... still no answers in here eh? If I figure it out I'll reply

**Comment by webboty** [comment, marked as answer]
Hi guys @adamhaley & @fractalbit 

... i was wondering that too and got it to work. 

Just take advantage of the noty message system backpack is using.
When you intercept the delete you can send back with a return of an array of a specific structure.

```
$custom_message['error'][0] = 'You super awesome message can go here. Of course you prepared it dynamically via php.';
return $custom_message;    
```
            
Instead of using error you can also use success in case it is successful etc..
If you have multiple bubbles you want to show ... just add more by increasing the counter at the end per type.

Here an example for multiple messages:

```
            $custom_message['error'][0] = 'Red Error Bubble 1';
            $custom_message['error'][1] = 'Red Bubble 2';
            $custom_message['success'][0] = 'This stuff works. Deleted.';
            $custom_message['warning'][0] = 'My warning here.';
``` 


Hint: When you intercept the destroy/delete procedure make sure you hand over the ID to the delete hook in case you want to delete the item after doing your checks. Or do not hand it over if you do not want to delete it ... and of course show the custom error message. 

Overall your CRUD function could look like this:

```
    //intercept delete operation
    protected function destroy($id)
    {
        //check if has access to delete operation
        $this->crud->hasAccessOrFail('delete');


       //do some stuff and or checks here

      //use delete of laravel (uses soft delete of laravel when added to the controller)
      $this->crud->delete($id);
      
      //return with custom message
      $custom_message['success'][0] = 'Yea! Stuff got deleted!';
      return $custom_message;    
    }
```


That's it ... worked for me.
Cheers

**Comment by jcastroa87** [comment]
I will close due to not activity, thanks @webboty for that answer.
